lib/socket-pool.js

```diff
@@ -0,0 +1,181 @@
+const udx = require('udx-native')
+
+module.exports = class SocketPool {
+  constructor (dht) {
+    this._dht = dht
+
+    this._sockets = new Set()
+    this._free = new Set()
```

Comment:
this should be an array imo so we can track causality later on

Reply:
What exactly would we track with an array?